active/NavigationView/HierarchicalNav/HierarchicalNavigationView.md

```diff
@@ -0,0 +1,354 @@
+    {
+        this.InitializeComponent();
+    }      
+    
+    public ObservableCollection<Category> Categories = new ObservableCollection<Category>()
+    {
+        new Category(){
+            Name = "Menu Item 1",
+            Icon = "Icon",
+            Children = new ObservableCollection<Category>() {
+                new Category(){
+                    Name = "Menu Item 2",
+                    Icon = "Icon",
+                    Children = new ObservableCollection<Category>() {
+                        new Category() { 
+                            Name  = "Menu Item 2", 
+                            Icon = "Icon",
+                            Children = new ObservableCollection<Category>() {
+                                new Category() { Name  = "Menu Item 3", Icon = "Icon", IsLeaf = true },
+                                new Category() { Name  = "Menu Item 4", Icon = "Icon", IsLeaf = true }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        new Category(){
+            Name = "Menu Item 5",
+            Icon = "Icon",
+            Children = new ObservableCollection<Category>() {
+                new Category(){
+                    Name = "Menu Item 6",
+                    Icon = "Icon",
+                    Children = new ObservableCollection<Category>() {
+                        new Category() { Name  = "Menu Item 7", Icon = "Icon", IsLeaf = true },
+                        new Category() { Name  = "Menu Item 8", Icon = "Icon", IsLeaf = true }
+                    }
+                }
+            }
+        },
+        new Category(){ Name = "Menu Item 9", Icon = "Icon", IsLeaf = true }
+    };
+}
+```
+
+Selected items will draw their selection indicators along their left edge when in left mode or their bottom edge when in top mode. 
+The selected item may not always remain visible.
+For example, the selected item may be a child node inside a non-expanded subtree.
+In this situation, the first visible ancestor of the selected item will show as selected, and the selection indicator will move as users expand the subtree. 
+The entire navigation view will show no more than one selection indicator.
+
+In both Top and Left modes, clicking the arrows on NavigationViewItems will expand or collapse the subtree. Clicking or tapping 
+_elsewhere_ on the NavigationViewItem will trigger the `ItemInvoked` event, and it will also collapse or expand the subtree.
+
+## Keyboarding
+Users can move focus around the navigation view using their [keyboard](https://docs.microsoft.com/en-us/windows/uwp/design/input/keyboard-interactions). 
+The arrow keys expose "inner navigation" within the pane and follow the interactions provided in [tree view](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view).
+- **Up arrow** moves focus to the item directly above the item currently in focus
```

Comment:
Presumably this changes slightly when the NavigationView is in Top mode.

Reply:
@ojhad?

Reply:
Yep, in top nav, the behavior should change for the primary list (the left and right keys are used to move focus between items). 
Navigation in top nav flyouts would be the same as in left nav.

Reply:
That makes sense to me. @ojhad is that how it's implemented?

Reply:
It's more complicated than that, since there is also the "in flyout/not in flyout" dimension. Something like

| Mode | Up/Down | Right/Left |
|-|-|-|
| Left, not in flyout | Movement | Expand and collapse |
| Top, not in flyout | Expand and collapse(?) | Movement |
| In flyout | Movement | Expand and collapse |

Reply:
How should top-mode handle expanding and collapsing children? In the flyout, it uses up and down to navigate between the children items. Hitting the down key on the parent item of the NavView will expand the list, but how should we go about collapsing it once you're on the first child?

Is tapping the up key twice an option? I don't think left or right would be as intuitive to open or close a vertical-list-flyout.

Reply:
> That makes sense to me. @ojhad is that how it's implemented?

In the current implementation, only invoking the item will expand/collapse. Arrow keys are only used for navigation.

> How should top-mode handle expanding and collapsing children?

Currently, the only way to close a flyout is either by hitting the esc key or by selecting a leaf node

Reply:
> how should we go about collapsing it once you're on the first child?
 
We could potentially detect if you are navigating up from the top most item and change focus that way. Though I don't know if we'd want to close the flyout in a similar manner when the last element is reached and the down key is pressed.

Reply:
I see. I think top-mode can be expanded by the down key on the parent element, and collapsed by hitting the esc key. I'll update the spec later today with all of these details.

Reply:
Here's what a description of the current behavior would be:

- **Up Arrow**
    - When in Left Nav or Flyout, moves focus to the item directly above the item currently in focus
    - When in Top Nav, does nothing

- **Down Arrow**
    - When in Left Nav or Flyout, moves focus the item directly below the item currently in focus; note that the items do not need to be visually adjecent, focus will move from the last item in the pane's list to the settings item.
    - When in Top Nav, does nothing

- **Right Arrow**
    - When in Top Nav, moves focus to the item directly to the right of the item currently in focus
    - When in Left Nav, does nothing

- **Left Arrow**
    - When in Top Nav, moves focus to the item directly to the left the item currently in focus
    - When in Left Nav, does nothing

- **Space/Enter Key**
    - Invokes/Selects item
    - When in Left Expanded mode and item has children, expands/collapses item
    - When in Left Compact/Top mode and item has children, expands children into a flyout and places focus on first item in flyout
    - When in flyout and on leaf node, invokes/selects item and closes flyout

- **Esc Key**
    - When in flyout, closes the flyout

Reply:
@MikeHillberg @oldnewthing @ojhad 
I just added these specifics into the spec to describe the current behavior. I've also opened an issue [#2297](https://github.com/microsoft/microsoft-ui-xaml/issues/2297) so in the future we can add support for arrow keys expanding and collapsing the children items.

However, since we want this to be ready for 2.4 we won't block on these changes as the current behavior leaves things fully functional.